pkg/build/nix_generic.go

```diff
@@ -0,0 +1,79 @@
+package build
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"reflect"
+
+	"github.com/testground/testground/pkg/api"
+	"github.com/testground/testground/pkg/rpc"
+)
+
+var (
+	_ api.Builder = &NixBuilder{}
+)
+
+// NixBuilder (id: "nix:generic") is a builder that build the test plan using
+// Nix. The resulting artifact can be a binary or a container.
```

Comment:
Do we care about distinguishing between binary/container artifacts? Should we flag it somehow?